### ISSUE TITLE
Bug: Sync error: updates to previously synchronised events are not received

### ISSUE THREAD

**Issue by cudeso** [MISP/MISP#9755]
### Actual behavior

Two MISP servers, both set as "local" instances. Owned by the same org_uuid.

Synchronisation rules are: 
- B pushes all with workflow:state=draft to A
- B pulls everything with workflow:state=complete from A
The underlying idea: "draft is created on B, send for validation to A, state is changed and then back to B".

An event is published on server B. It has attributes, global tags and a local tag :"workflow:state=draft". The event is pushed to A and shows up in the event list.
On server A, the local tag workflow:state=draft is removed, and a new local tag workflow:state=complete is added. The event is then published. Nothing else is changed. The pull rule on B (to pull from A) ignores the updated event, no clear error indication. Tried with distribution settings All communities and Connected Communities.

If you create an event first on A with workflow:state=complete it's pulled by B from A without problems.

### Expected behavior

Server B pulls in the "updates" for the event done on server A.

### Steps to reproduce

See actual behaviour./

### Version

2.4.188

### Operating System

Ubuntu

### Operating System version

20.04

### PHP version

7.4

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by iglocska** [comment]
Not a bug, but a feature. B rejects all changes to its own events from other instances, that's what the locked field is for.

However, I see why this can be problematic in multi misp internal workflows, maybe we could relax the feature in the future for internal syncs?

**Comment by cudeso** [comment]
Ha! Thanks!
Yes, definitely, there's a use for it for internal syncs.
To be on the safe side, maybe as a configurable option. "MISP.ignore_lock_internalinstance"?

**Comment by cudeso** [comment]
Also maybe linked to https://github.com/MISP/MISP/issues/4920

**Comment by cudeso** [comment]
Hi @iglocska . Do you know if there any plans to implement a more relaxed approach for internal syncs (ie., something 'ignore_lock_internalinstance' or similar)? Just checking for the progress, otherwise I'll continue with an alternative approach via tags to achieve something similar.

**Comment by iglocska** [comment]
Good point. Not entirely sure how straightforward it would be, but indeed, internal instance sync should ignore the locking mechanism by default, silly that we don't have an exception for it...

**Comment by iglocska** [comment]
Need to remember this / spend some time on it, if you don't see it in the near future can you ping me offline and gently remind me (=kick my arse if I don't do it)?

**Comment by cudeso** [comment]
Good ;-) 
In the mean time I have this workaround with tags to force some review on first server ('B' or 'Internal'), and then also a second review on second server ('A' or 'Edge').
![curation-share-with-external-communities](https://github.com/user-attachments/assets/3bcf7cc0-c447-4f8f-9b84-818364d59142)
